pkg/resources/secret_with_oauth_authorization_code_grant.go

```diff
@@ -146,9 +147,11 @@ func ReadContextSecretWithAuthorizationCodeGrant(withExternalChangesMarking bool
 			return diag.FromErr(err)
 		}
 
-		if withExternalChangesMarking {
-			if err = handleExternalValueChangesToObjectInDescribe(d,
-				describeMapping{"oauth_refresh_token_expiry_time", "oauth_refresh_token_expiry_time", secretDescription.OauthRefreshTokenExpiryTime.String(), secretDescription.OauthRefreshTokenExpiryTime.String(), nil},
+		// if secret type is changed externally, we wont be able to read oauth_refresh_token_expiry_time value (since it will not be provided)
```

Comment:
I think in such cases, we should check if we can return early (immediately after show/desc and setting the type). We're setting some values here, but the resource is going to be recreated anyway in this scenario.

Reply:
im not really sure how to check it and return earlier in a way that checking the type change for same `OAUTH2` type will work... we relay on describe_output in that case, so it needs to be set either way

Reply:
That's okay, I thought it would be simpler :/